r/src/compute.cpp

```diff
@@ -264,6 +264,12 @@ std::shared_ptr<arrow::compute::FunctionOptions> make_compute_options(
                                      max_replacements);
   }
 
+  if (func_name == "day_of_week") {
+    using Options = arrow::compute::TemporalComponentExtractionOptions;
+    return std::make_shared<Options>(cpp11::as_cpp<bool>(options["one_based_numbering"]),
```

Comment:
Looks good, although 1 minor suggestion - please could we have default options set here so if someone calls `day_of_week` via `call_function("day_of_week", x)`, they don't have to supply either argument?

Reply:
Sure!

Reply:
https://github.com/apache/arrow/pull/10598/commits/bf4ab293a4353e7e357030dc5675baedb042e5ec - does this look right?

Reply:
@rok Not quite - `one_based_numbering` has a default value of `false` in C++, whereas in R it should be `true`.  I _think_ the way to achieve it is to not use the C++ defaults and do something like this: https://github.com/apache/arrow/blob/e9fa30406215b76ed6c885302fbfe6075c47badf/r/src/compute.cpp#L244-L254

Reply:
Got it, you mean like this?
```C++
if (func_name == "day_of_week") {
  using Options = arrow::compute::TemporalComponentExtractionOptions;
  bool one_based_numbering = true;
  return std::make_shared<Options>(one_based_numbering,
                                   cpp11::as_cpp<uint32_t>(options["week_start"]));
}
```

Reply:
https://github.com/apache/arrow/pull/10598/commits/3539634894c962f463b3947c03b9e0be135061ee

Reply:
I think that's OK - I had in mind also adding in 
```r
if (!Rf_isNull(options["one_based_numbering"])) { 
     one_based_numbering = cpp11::as_cpp<bool>(options["one_based_numbering"]); 
 } 
```
but now I think it might be redundant, so your solution is fine!

Reply:
Would users ever use it? If yes then I'd add it.

Reply:
I think it'd be very uncommon or never used at all, and even if someone wanted to use it, it'd be easy for them just to deduct 1 from whatever value they get, so I don't think it matters.

Reply:
I added your suggestion just in case :).